ci_example_cpp/doc/cpp_coding_guidelines.md

```diff
@@ -0,0 +1,270 @@
+- Preferably each class should be in a separate file with name
+    *class_name_in_lower_case*.  However, this is not a strict rule, if
+    several smaller classes are logically closely related, they may go to
+    the same file.
+- Executable scripts should be placed in the `scripts/` folder. And should have
+    a CMake **install rule** that makes them executable upon installation.
+- The C++/pybind11 code for wrapping C++ code to Python must be placed in
+    `srcpy/`
+
+## III. Naming
+
+Give as descriptive a name as possible, within reason. Do not worry about saving
+horizontal space as it is far more important to make your code immediately 
+understandable by a new reader. Do not use abbreviations that are ambiguous or 
+unfamiliar to readers outside your project, and do not abbreviate by deleting 
+letters within a word. Abbreviations that would be familiar to someone outside 
+your project with relevant domain knowledge are OK. As a rule of thumb, an 
+abbreviation is probably OK if it's listed in Wikipedia.
+
+Formatting of names should be as follows:
+
+- types (classes, structs, ...): *FirstUpperCamelCase*
+- functions, methods: *lower_case_with_underscores*
+- variables: *lower_case_with_underscores*
+- class members: *like_variables_but_with_trailing_underscore_*
+- constants: *UPPER_CASE_WITH_UNDERSCORES*
+- global variables: Should generally be avoided but if needed, prefix them with
+    g_, i.e. *g_variable_name*.
+
+## IV. Add Units to Variable Names
+
+Variables that hold values of a specific unit should have that unit appended to
+the name.  For example if a variable holds the velocity of a motor in *krpm* it
+should be called `velocity_krpm` instead of `just velocity`. Some more examples:
+
+- duration_us (use "u" instead of "µ")
+- voltage_mV
+- acceleration_mps2 (\f$ \frac{m}{s^2} \f$)
+
+## V. C/C++ Formatting
+
+### V.1. Line Length
+
+Limit the length of lines to 80 characters.
+
+This may seem hard to follow sometimes but makes it much easier to view two or even three files next to each other (important during code review or when resolving merge conflicts).
+
+### V.2. Indentation
+
+- Use spaces instead of tabs.
+- 4 spaces per "tab".
+
+### V.3. Position of braces
+
+- Opening brace always goes to the next line.
+- **Always** add braces for single-line if/loop/etc.
+
+Example:
+
+~~~{.c}
+namespace bar
+{
+
+class Foo {
+    void my_function(const Foo &foo, int *output_arg)
+    {
+        if (condition)
+        {
+            ...
+        }
+        else if (other_condition)
+        {
+            ...
+        }
+        else
+        {
+            ...
+        }
+
+        while (condition)
+        {
+            ...
+        }
+    }
+}
+
+} // namespace
+~~~
+
+### V.3. Spaces
+
+Add single spaces between if/for/etc., the condition and the brace. add spaces
+around most binary operators. Exception: No spaces around *::*, *.* and *->*.
+Also no spaces for unary operators (*i++*, *&x*, **x*, ...).
```

Comment:
I think using formatted text instead of italic for thinks that refer to code makes it a bit more readable (e.g. `i++` instead of *i++*)

Reply:
so \`*x\` ?

Reply:
Sorry, I missed that one. Yes.